clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -2682,19 +2704,53 @@ void SYCLIntegrationHeader::emit(raw_ostream &O) {
   }
   O << "};\n\n";
 
+  O << "// array representing kernel parameters used in all kernels defined in "
+       "the\n";
+  O << "// corresponding source\n";
+  O << "static constexpr\n";
+  O << "const bool kernel_param_used[] = {\n";
```

Comment:
The backend implement the 'reverse' logic here.  So call it param_omit_table.
The comments need to be: OMIT_TABLE_BEGIN and OMIT_TABLE_END